modules/front-end/src/app/core/components/header/header.component.ts

```diff
@@ -91,25 +84,31 @@ export class HeaderComponent implements OnInit {
     return this.currentProjectEnv?.envId === env.id;
   }
 
-  canListProjects = false;
-
   get availableProjects() {
-    return this.canListProjects ? this.allProjects : [];
+    return this.allProjects;
   }
 
-  canListEnvs = false;
-
   get availableEnvs() {
     const project = this.allProjects.find(x => x.id === this.selectedProject.id);
-    return this.canListEnvs ? project?.environments : [];
+    if (!project) {
+      return [];
+    }
+
+    return project.environments.filter((env) => {
+      const envRN = this.permissionsService.getEnvRN(project.name, env.name);
+      return !this.permissionsService.isDenied(envRN, permissionActions.CanAccessEnv);
+    });
```

Comment:
Needn't check permission again since all envs in `this.allProjects` has been checked before in  `setAllProjects` function.

Reply:
done